tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+		RedefineDefaultCMD([]string{"/bin/bash", "-c",
+			fmt.Sprintf("testcmd -interface br-ex -protocol tcp -port %d -listen", portNum)}).
+		WithPrivilegedFlag().CreateAndWaitUntilRunning(180 * time.Second)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create nginx test pod")
+
+	return testPod
+}
+
+func validateTCPTraffic(clientPod *pod.Builder, destIPAddrs []string, interfaceName,
+	containerName string) error {
+	for _, destIPAddr := range removePrefixFromIPList(destIPAddrs) {
+		command := fmt.Sprintf("testcmd -interface %s -protocol tcp -port %d -server %s", interfaceName,
+			8888, destIPAddr)
+		_, err := clientPod.ExecCommand([]string{"bash", "-c", command}, containerName)
+
+		return err
+	}
+
+	return nil
+}
+
+func createMCAndWaitforMCPStable(fileContentString string) {
+	truePointer := true
+	stringgzip := "gzip"
+
+	mode := 384
+	fileContents := fileContentString
+	sysDContents := `
+            [Unit]  
+            Description=Netfilter Tables
+            Documentation=man:nft(8)
+            Wants=network-pre.target
+            Before=network-pre.target
+            [Service]
+            Type=oneshot
+            ProtectSystem=full
+            ProtectHome=true
+            ExecStart=/sbin/nft -f /etc/sysconfig/nftables.conf
+            ExecReload=/sbin/nft -f /etc/sysconfig/nftables.conf
+            ExecStop=/sbin/nft 'add table inet custom_table; delete table inet custom_table'
+            RemainAfterExit=yes
+            [Install]
+            WantedBy=multi-user.target`
+	ignitionConfig := ignition.Config{
+		Ignition: ignition.Ignition{
+			Version: "3.4.0",
+		},
+		Systemd: ignition.Systemd{
+			Units: []ignition.Unit{
+				{
+					Enabled:  &truePointer,
+					Name:     "nftables.service",
+					Contents: &sysDContents,
+				},
+			},
+		},
+		Storage: ignition.Storage{
+			Files: []ignition.File{
+				{
+					Node: ignition.Node{
+						Overwrite: &truePointer,
+						Path:      "/etc/sysconfig/nftables.conf",
+					},
+					FileEmbedded1: ignition.FileEmbedded1{
+						Contents: ignition.Resource{
+							Compression: &stringgzip,
+							Source:      &fileContents,
+						},
+						Mode: &mode,
+					},
+				},
+			},
+		},
+	}
+	finalIgnitionConfig, err := json.Marshal(ignitionConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to serialize ignition config")
+	_, err = mco.NewMCBuilder(APIClient, mcNftablesName).
+		WithLabel("machineconfiguration.openshift.io/role", NetConfig.CnfMcpLabel).
+		WithRawConfig(finalIgnitionConfig).
+		Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create nftables machine config")
+
+	err = WaitForMcpStable(APIClient, 4*time.Minute, 30*time.Second, "workercnf")
+	Expect(err).ToNot(HaveOccurred(), "Failed to wait for MCP to be stable")
+}
+
+// WaitForMcpStable waits for the stability of the MCP with the given name.
+func WaitForMcpStable(apiClient *clients.Settings, waitingTime, stableDuration time.Duration, mcpName string) error {
```

Comment:
I'm pretty sure we have the same func somewhere, please reuse it

Reply:
Found it